### ISSUE TITLE
DNSSEC/dnssec03 and DNSSEC/dnssec05 TC crash in case of "empty" keys

### ISSUE THREAD

**Issue by vlevigneron** [zonemaster/zonemaster-ldns#110]
Here is a domain with zero length RSA key
```
nlagriculture.nl.	86400 IN DNSKEY	257 3 13 (
				vRMOgGXuo/RaeD1XNWmDPvSDQDCNRa68OTq+/BqucA25
				ACppkunDt5fnRF2IL76ZpybyG7W8IBNYj7dpYrzWOg==
				) ; KSK; alg = ECDSAP256SHA256 ; key id = 12993
nlagriculture.nl.	86400 IN DNSKEY	256 3 8 (
				AwEAAeBjJKDZyqH6QoTM3lSHZ2Gk+26y6S/MH1Va2noA
				MfGAu1t2jg8Ve1m1dR53hNjaQpDiKkwg9UNH7wCdEKPT
				/PoAMP3Q3UZnqnoPk4b2wwyGEN5v+eIvDREoc6Eg1leA
				A8RQ2/GatjqWVR5K7wQ1UFCSyIOhgiPYpOKqfoFAnmx1
				) ; ZSK; alg = RSASHA256 ; key id = 18111
nlagriculture.nl.	86400 IN DNSKEY	257 3 8 (
				
				) ; KSK; alg = RSASHA256 ; key id = 1033
nlagriculture.nl.	86400 IN DNSKEY	256 3 8 (
				AwEAAfc58Rv7aL6+1cVlGaGF6P5mjxE77hDFXAKJFukO
				prucKqO1zupdHGDHrkWlvnBYxR+MArXPqZxsXAjNNgdb
				HkCcKG4S+qho+o1hjalGiTmslsdKCTlFoSf7u6Q4rlVf
				r/yUbF09cQdCMdjBEjcnezRBSAUNa7zu6fPPDdZJ/tfj
				) ; ZSK; alg = RSASHA256 ; key id = 61578
```
both test cases crash with the following message

```
   2.62 INFO      The DNSKEY with tag 61578 uses algorithm number 8 (RSA/SHA-256), which is OK.
   2.62 DEBUG     Key with keytag 61578 details : Size = 1024, Flags (SEP bit *not* set, RFC 5011 revocation bit *not* set).
   2.62 INFO      The DNSKEY with tag 18111 uses algorithm number 8 (RSA/SHA-256), which is OK.
   2.62 DEBUG     Key with keytag 18111 details : Size = 1024, Flags (SEP bit *not* set, RFC 5011 revocation bit *not* set).
   2.62 INFO      The DNSKEY with tag 1033 uses algorithm number 8 (RSA/SHA-256), which is OK.
perl: ./rdata.c:40: ldns_rdf_data: Assertion `rd != NULL' failed.
```

**Comment by matsduf** [comment]
> Well, it could be, even if I'm not sure the behaviour of the Engine would be what we expect. 

Do I understand it correctly that you say that Engine does not use the Zonemaster-LDNS methods correctly? Do we need to document the Zonemaster-LDNS API better, make it more well-defined?

**Comment by vlevigneron** [comment]
@matsduf no but as this is something we never met in the past (I mean this kind of strange DNSKEY record), I can not say that the Engine will give the error we expect for sure. It's something we will have to test once it will be fixed in LDNS and/or Zonemaster-LDNS.

**Comment by matsduf** [comment]
We have no test case checking for zero lenght keys. I did not even think it was possible. It is a bug, but not likely to happen, so the question is how much attention we should give it.

**Comment by matsduf** [comment]
This issue includes checking for updates on upstream.

**Comment by mattias-p** [comment]
I failed to reproduce this when running `zonemaster-cli --no-ipv6 nlagriculture.nl`. The zone has likely been fixed since this error was reported.

I also tried to reproduce it by letting Zonemaster::LDNS parse a synthesized package in wireformat with an empty DNSKEY record. This didn't crash either, so it seems the problem isn't in the parser.

**Comment by mattias-p** [comment]
I managed to reproduce the crash like this:
```sh
zonemaster-cli --no-ipv6 nlagriculture.nl. --test dnssec/dnssec03 --restore data.txt
```
[data.txt](https://github.com/zonemaster/zonemaster-ldns/files/8203735/data.txt)

**Comment by matsduf** [comment]
Resolved by #135.
